### PR TITLE
Renaming MKL-DNN as DNNL

### PR DESCRIPTION
**Pull request by sreekanth-yalachigere**
Reason
To simplify library naming and differentiate it from Intel MKL, starting with version 1.1 MKL-DNN library name is changed to Deep Neural Network Library (DNNL).

### CODE REVIEW

**Comment by jywu-msft** [comment]
you need to update the CI yml files under tools/ci_build/github/azure-pipelines to use the renamed build option. 
can you grep through the source tree and double check where --use_mkldnn was being used?

File: onnxruntime/test/onnx/main.cc [hunk 1]
```diff
@@ -356,14 +356,23 @@ int real_main(int argc, char* argv[], Ort::Env& env) {
...
+                                                "tf_resnet_v2_101", "tf_resnet_v2_152"};
+	
     std::unordered_set<std::string> all_disabled_tests;
     if (enable_cuda) {
       all_disabled_tests.insert(std::begin(cuda_flaky_tests), std::end(cuda_flaky_tests));
     }
     if (enable_dml) {
       all_disabled_tests.insert(std::begin(dml_disabled_tests), std::end(dml_disabled_tests));
     }
+    if (enable_mkl) {
```
Comment (jywu-msft) [inline onnxruntime/test/onnx/main.cc]: rename enable_mkl to enable_dnnl ?

**Comment by sreekanth-yalachigere** [comment]
> you need to update the CI yml files under tools/ci_build/github/azure-pipelines to use the renamed build option.
> can you grep through the source tree and double check where --use_mkldnn was being used?

updated all occurrences.

**Comment by jywu-msft** [comment]
some onnx tests failed in Linux CPU CI? can you take a look? 

test averagepool_2d_ceil failed, please fix it
test maxpool_2d_ceil failed, please fix it
test maxpool_2d_dilations failed, please fix it

**Comment by jywu-msft** [comment]
> some onnx tests failed in Linux CPU CI? can you take a look?
> 
> test averagepool_2d_ceil failed, please fix it
> test maxpool_2d_ceil failed, please fix it
> test maxpool_2d_dilations failed, please fix it

the same tests fail in python.
please disable them in https://github.com/microsoft/onnxruntime/blob/9441122bd1aaeeed81a0f5f1a702a30061700a5b/onnxruntime/test/python/onnx_backend_test_series.py#L143
note the name of the tests are slightly different (see below)

FAIL: test_averagepool_2d_ceil_cpu (__main__.OnnxBackendNodeModelTest)
FAIL: test_maxpool_2d_ceil_cpu (__main__.OnnxBackendNodeModelTest)
FAIL: test_maxpool_2d_dilations_cpu (__main__.OnnxBackendNodeModelTest)

**Pull request merged**